packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -37,7 +48,7 @@ export const withReactQuery = (Wrapped) => {
          */
         static async doInitAppState({res, appJSX}) {
             const queryClient = (res.locals.__queryClient =
-                res.locals.__queryClient || new QueryClient())
+                res.locals.__queryClient || new QueryClient(queryClientConfig))
 
             await ssrPrepass(appJSX)
```

Comment:
Is it possible for the prepass to work on **_multiple_ query clients**? I'm thinking: what if this code block deals with more than 1 `queryClient`..

From react-query angle, they said it is possible to work with [multiple clients/providers](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#custom-contexts-for-multiple-providers). Essentially:
1. Create a new React context
2. Pass the above context to the hook calls and the query client provider
3. Repeat for every package/module
3. Now when making a query, it will be associated with the correct client provider

(Note: we talked previously about using a different option called `contextSharing`. But it looks like that option is no longer relevant? That's what one of the main contributors of react-query said in this comment: https://github.com/TanStack/query/issues/2810#issuecomment-1213366716)

Now the question is whether we can support multiple clients on our end or not.

Reply:
Just the close out this comment. The PWA-Kit SDK for now will only support a single react query client/provider. This is something that we'll look for feedback from partners to see if there is room for improvement in the future.